stl/inc/xutility

```diff
@@ -5808,32 +5808,30 @@ _NODISCARD _CONSTEXPR20 _FwdIt lower_bound(_FwdIt _First, _FwdIt _Last, const _T
 // FUNCTION TEMPLATE _Swap_ranges_unchecked
 template <class _FwdIt1, class _FwdIt2>
 _CONSTEXPR20 _FwdIt2 _Swap_ranges_unchecked(_FwdIt1 _First1, const _FwdIt1 _Last1, _FwdIt2 _First2) {
-    // swap [_First1, _Last1) with [_First2, ...), no special optimization
-    for (; _First1 != _Last1; ++_First1, (void) ++_First2) {
-        _STD iter_swap(_First1, _First2);
-    }
-
-    return _First2;
-}
+    // swap [_First1, _Last1) with [_First2, ...)
 
 #if _USE_STD_VECTOR_ALGORITHMS
-template <class _Ty, enable_if_t<_Is_trivially_swappable_v<_Ty>, int> = 0>
-_CONSTEXPR20 _Ty* _Swap_ranges_unchecked(_Ty* _First1, _Ty* const _Last1, _Ty* _First2) {
-    // swap [_First1, _Last1) with [_First2, ...), trivially swappable optimization
+    using _Elem1 = remove_reference_t<_Iter_ref_t<_FwdIt1>>;
+    using _Elem2 = remove_reference_t<_Iter_ref_t<_FwdIt2>>;
+    if constexpr (is_same_v<_Elem1, _Elem2> && _Is_trivially_swappable_v<_Elem1> && //
+                  _Iterators_are_contiguous<_FwdIt1, _FwdIt2>) {
 #ifdef __cpp_lib_is_constant_evaluated
-    if (_STD is_constant_evaluated()) {
-        for (; _First1 != _Last1; ++_First1, (void) ++_First2) {
-            _STD iter_swap(_First1, _First2);
+        if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            __std_swap_ranges_trivially_swappable_noalias(
+                _To_address(_First1), _To_address(_Last1), _To_address(_First2));
+            return _First2 + (_Last1 - _First1);
         }
+    }
+#endif // _USE_STD_VECTOR_ALGORITHMS
 
-        return _First2;
+    for (; _First1 != _Last1; ++_First1, (void) ++_First2) {
+        _STD iter_swap(_First1, _First2);

```

Comment:
I think we can inline this `_STD iter_swap` call as `_Swap_adl(*_First1, *_First2)` now that users are forbidden to specialize function templates in namespace `std`.

Reply:
AFAIC `iter_swap` calls `swap` directly (just like `_Swap_adl`) so there will be no improvement.
https://github.com/microsoft/STL/blob/3a279357e05a70278cb5114143ab798f59b196b0/stl/inc/utility#L78-L82

Reply:
Fair enough, my suggestion simply trades one unnecessary function template instantiation for another. Let's leave this as-is, and if I want to convince folks that we should be writing `swap(meow, woof); // swap intentionally unqualified ` directly instead of using `_Swap_adl` and `iter_swap` I can do so in another PR.